src/ucp/wireup/select.c

```diff
@@ -806,6 +807,32 @@ ucp_wireup_ep_init_flags(const ucp_wireup_select_params_t *select_params,
     return select_params->ep_init_flags | select_ctx->ucp_ep_init_flags;
 }
 
+static ucs_status_t
+ucp_wireup_add_cm_lane(const ucp_wireup_select_params_t *select_params,
+                       ucp_wireup_select_context_t *select_ctx)
+{
+    ucp_wireup_select_info_t select_info;
+
+    if (!(select_params->ep_init_flags & (UCP_EP_INIT_CM_WIREUP_CLIENT |
+                                          UCP_EP_INIT_CM_WIREUP_SERVER))) {
+        return UCS_OK;
+    }
+
+    select_info.reachable  = 1;  /**< UCT CM provides reachability by its API
+                                      CBs */
+    select_info.priority   = 0;  /**< Currently we have only 1 CM
+                                      implementation */
+    select_info.rsc_index  = UCP_NULL_RESOURCE; /**< RSC doesn't matter for CM */
```

Comment:
are all wireup function capable of handling such rsc_index?
can we avoid using ucp_wireup_add_lane_desc() for cm_lane? just add it separately, like wireup lane

Reply:
Then we can't reuse common selection logic when add support of more 1 CMs. It's based on lane_desc.

Reply:
i think the selection logic should be for transport lanes and not CM lanes
it's different because transport lanes can be reused for multiple purposes (am/rma/..) so need to avoid duplications
while cm lane is separate purpose which is never same lane as am/rma/amo/...

Reply:
as discussed offline - keep using select_info/lane_desc as is because CMs also require selection logic